browser/brave_ads/ads_service_impl.cc

```diff
@@ -2100,7 +2100,7 @@ void AdsServiceImpl::RecordP2AEvent(const std::string& name,
         break;
       }
 
-      for (auto& item : list->GetList()) {
+      for (auto& item : list->GetListDeprecated()) {
```

Comment:
do we have a follow-up issue to not use the deprecated method anymore?

Reply:
I'm happy to create an issue for this, but according to https://chromium.googlesource.com/chromium/src/+/354945de1fb564ef04c07cf8bfedf434d2d81747 it sounds like `GetList` is going to be reimplemented at some still-unknown point in the future. We were thinking that when that happens, the deprecated version will be removed and we'll have to fix this at that time anyway.

Reply:
weird, I guess it's fine now

Reply:
It seems we hit a timing issue here: @emerick and I talked on February 15th about this and at that point the activity in the CLs associated to https://bugs.chromium.org/p/chromium/issues/detail?id=1291666 did indeed transmit the idea that upstream was not yet ready to migrate away of `GetListDeprecated()`, so we followed suit and did the same here.

But I just checked now and it seems that right after we talked they landed https://chromium-review.googlesource.com/c/chromium/src/+/3443197 and https://chromium-review.googlesource.com/c/chromium/src/+/3472571, so it seems we could actually migrate this now.

I'll try to do that migration today/tomorrow and put it on top of cr100, if I can't I'll file a issue to track the effort to be done on top of master once this branch gets merged. 

Apologies for the confusion, we should have checked the upstream bug but at the time it did indeed felt like the actual possibility of migration was further away than what it actually was in the end.

Reply:
Tentatively fixed in https://github.com/brave/brave-core/commit/1807e9ee3a0ee9c57e8f0f57f6fba6b2db940f90 !

I say "tentatively" because so far I could only check locally that it compiles fine and that all unit and browser tests pass correctly **on Linux**, but it's possible some small adjustment is needed on Windows/Mac/Android as well, although this should be the bulk of it.

PS: I'm also building on Mac in parallel and so far no issues

Reply:
My local build on Mac also went fine: compiled correctly and passed all tests

Reply:
Somehow I missed a few migrations yesterday, so here they are as a fixup: https://github.com/brave/brave-core/commit/1391f872ed0f02f492957c022cccc5144034086c